constants/sprite_anim_constants.asm

```diff
@@ -70,6 +70,7 @@ DEF NUM_SPRITEANIMDICT_ENTRIES EQU 10
 	const SPRITE_ANIM_INDEX_DEX_SLOWPOKE
 	const SPRITE_ANIM_INDEX_TOWN_MAP_FLY
 	const SPRITE_ANIM_INDEX_GREEN_WALK
+	const SPRITE_ANIM_INDEX_FLY_MON
```

Comment:
If this whole "separate sprite anim for Fly mon" change is independent of the weather feature, can it possibly be refactored into its own PR? That would make its effects easier to test independently of weather, and would clarify in the commit history which changes are for what.

Reply:
Not really... The FlyFrom and FlyTo animations were reworked in weather. Before `FlyFrom` would clear all objects on the screen before starting the fly from animation. Now we are only clearing all objects on the screen if we run out of OAM slots for the fly animation, or if we out of palette slots for it. Since the palette slot is now "dynamic", we needed a sperate sprite animation to handle this instead of `RED_WALK` which always used Obj Slot 0. These changes are fairly dependent on the new weather system's way of loading objects and sprites. Could I do this change after this PR? maybe.. but i'd prefer not to.  
  
Most of _this_ change came in the commit titled `Big Fly rework and others`.